lib/omniauth/strategies/bolt.rb

```diff
@@ -39,14 +39,26 @@ def request_phase
 
       def callback_phase
         payload = {
-          grant_type: 'authorization_code',
-          code: session['authorization_code'],
           client_id: options['publishable_key'],
-          # scope: 'openid+bolt.account.manage',
-          scope: session['scope'],
           client_secret: options['api_key']
         }
 
+        token_and_scope = if session['refresh_token']
+                            {
+                              grant_type: 'refresh_token',
+                              refresh_token: session['refresh_token'],
+                              scope: session['refresh_token_scope']
+                            }
+                          else
+                            {
+                              grant_type: 'authorization_code',
+                              code: session['authorization_code'],
+                              scope: session['scope']
```

Comment:
From now these session variables won't be used anymore. 
What do you think to delete them from the session?

Reply:
reworked logic in https://github.com/nebulab/omniauth-bolt/pull/2/commits/9b3154e439f4f511d1c1335517c5b9ba666567ab

Reply:
fixed in https://github.com/nebulab/omniauth-bolt/pull/2/commits/986cd543eb3c2d65eec5b70f9092820715ee2c58